### DISCUSSION TITLE
List plugins capabilities/accessibilities

### DISCUSSION THREAD

**Discussion by ghost** [lapce/lapce#2178] [Ideas]
For instance, I would like to know if a plugin access (or tries to access) the internet, I wouldn't want the plugin to download something fishy from the internet, or to have any connection at all.

Another example, not harmful but undesired for me and others, from the lapce-zig plugin: https://github.com/syntacti/lapce-zig/issues/3, I wouldn't know this if not for the issue or digging around the source. Of course, one can always request for better documentation or contribute to it, but that's not the point.

I think the list of access would feasible to extract from what APIs the Plugin calls. From the top of my head, I can think of: file system and internet. 

What do you think?

**Comment by panekj** [comment]
This has been discussed on our Discord already and there are plans for it

**Reply by ghost** [reply]
Oh, ok then. Thanks for the quick reply!
